### PR TITLE
faas-cli: 0.14.2 -> 0.14.3

### PR DESCRIPTION
**Pull request by techknowlogick**
###### Description of changes

https://github.com/openfaas/faas-cli/releases/tag/0.14.3

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/faas-cli/default.nix [hunk 1]
```diff
@@ -10,14 +10,14 @@ in
 buildGoModule rec {
   pname = "faas-cli";
   # When updating version change rev.
-  version = "0.14.2";
+  version = "0.14.3";
   rev = "b1c09c0243f69990b6c81a17d7337f0fd23e7542";
```
Comment (welteki) [inline pkgs/development/tools/faas-cli/default.nix:14]: Could you also update `rev` as stated in code comment.
```suggestion
  rev = "56b1a7db771a4f2062ec8d57a782d27a039ba9eb";
```

Comment (SuperSandro2000) [reply pkgs/development/tools/faas-cli/default.nix:14]: or we just drop that part. It is informational anyway.

Comment (welteki) [reply pkgs/development/tools/faas-cli/default.nix:14]: Yes, you are right. This is inconvenient. Let's do the same thing we did with arkade and k3sup en just hardcode the rev to `ref/tags/${version}`

Comment (techknowlogick) [reply pkgs/development/tools/faas-cli/default.nix]: thanks for suggestion :) I've just updated it

File: pkgs/development/tools/faas-cli/default.nix [hunk 2]
```diff
@@ -10,14 +10,14 @@ in
 buildGoModule rec {
   pname = "faas-cli";
   # When updating version change rev.
```
Comment (welteki) [inline pkgs/development/tools/faas-cli/default.nix:12]: Could you also remove this now obsolete comment.

File: pkgs/development/tools/faas-cli/default.nix [hunk 3]
```diff
@@ -10,14 +10,14 @@ in
...
-  version = "0.14.2";
-  rev = "b1c09c0243f69990b6c81a17d7337f0fd23e7542";
+  version = "0.14.3";
+  rev = "ref/tags/${version}";
```
Comment (welteki) [inline pkgs/development/tools/faas-cli/default.nix:14]: Maybe it would be better to inline this in line 31 now.

**Review by welteki — APPROVED** [review]

**Pull request merged**